Jenkinsfile

```diff
@@ -2,61 +2,438 @@
+						name 'DISTRO'
+						values  'empty', //NOTE: for correct draw in OpenBlueOcean
+							'DEB', 'RPM',
+							'debian8', 'debian9', 'debian10', 'debian11', 'debian12',
+							'amazon2', 'amazon2023',
+							'centos7', 'centos8', 'centos9',
+							'alma8', 'alma9',
+							'fedora31', 'fedora32', 'fedora34', 'fedora35', 'fedora36', 'fedora37', 'fedora38',
+							'ubuntu2004', 'ubuntu2204'
 					}
 				}
-				stage('RPM')
+				stages
 				{
-					steps
+					stage('DEB')
 					{
-						sh "bash ./build.sh ${env.BUILD_NUMBER} rpm"
-						deployRpm dir: "build-results_rpm", map_repo: pkg_map_branches('ootpa'), user: "rbrepo", agent: "agent"
-						deployRpm dir: "build-results_rpm", map_repo: pkg_map_branches('maipo'), user: "rbrepo", agent: "agent"
-						deployRpm dir: "build-results_rpm", map_repo: pkg_map_branches('plow'), user: "rbrepo", agent: "agent"
-						uploadArtifacts files: "build-results_rpm/*.rpm", dst: "${artifactoryRoot}", postfix: "RPM", shortnames: true, retention : false
+						when { expression { env.DISTRO == "DEB" } }
+						steps
+						{
+							node('dr-linbuild')
+							{
+								sh "bash ./build.sh ${env.BUILD_NUMBER} deb"
+								deployDeb dir: "build-results_deb", map_repo: pkg_map_branches('focal-agent'), user: "rbrepo", agent: "rep-agent"
+								deployDeb dir: "build-results_deb", map_repo: pkg_map_branches('jammy-agent'), user: "rbrepo", agent: "rep-agent"
+								deployDeb dir: "build-results_deb", map_repo: pkg_map_branches('bionic-agent'), user: "rbrepo", agent: "rep-agent"
+								deployDeb dir: "build-results_deb", map_repo: pkg_map_branches('bullseye-agent'), user: "rbrepo", agent: "rep-agent"
+									deployDeb dir: "build-results_deb", map_repo: pkg_map_branches('buster-agent'), user: "rbrepo", agent: "rep-agent"
+								deployDeb dir: "build-results_deb", map_repo: pkg_map_branches('bookworm-agent'), user: "rbrepo", agent: "rep-agent"
+								uploadArtifacts files: "build-results_deb/*.deb", dst: "${artifactoryRoot}", postfix: "DEB", shortnames: true, retention : false
+							}
+						}
+					}
+
+					stage('RPM')
+					{
+						when { environment name: 'DISTRO', value: 'RPM' }
+						steps
+						{
+							node('dr-linbuild')
+							{
+								sh "bash ./build.sh ${env.BUILD_NUMBER} rpm"
+								deployRpm dir: "build-results_rpm", map_repo: pkg_map_branches('ootpa'), user: "rbrepo", agent: "agent"
+								deployRpm dir: "build-results_rpm", map_repo: pkg_map_branches('maipo'), user: "rbrepo", agent: "agent"
+								deployRpm dir: "build-results_rpm", map_repo: pkg_map_branches('plow'), user: "rbrepo", agent: "agent"
+								uploadArtifacts files: "build-results_rpm/*.rpm", dst: "${artifactoryRoot}", postfix: "RPM", shortnames: true, retention : false
+							}
+						}
+					}
+
+					stage('Tests')
+					{
+						when {
+							allOf {
+								expression { env.DISTRO != "DEB" }
+								expression { env.DISTRO != "RPM" }
+								expression { env.DISTRO != "empty" }
```

Comment:
Since the building process for elastio is time-consuming let's confine conditions to run builds by PRs, develop, release, and build branches.